File: Java/Cyclic.java
package Java;

import java.util.Arrays;

public class Cyclic {
    public static void main(String[] args) {
        int[] arr = { 4, 3, 2, 1 };
        sort(arr);
        System.out.println(Arrays.toString(arr));

    }

    static void sort(int[] arr) {
        int i = 0;
        while (i < arr.length) {
            System.out.println(Arrays.toString(arr));
            int correct = arr[i] - 1;
            if (arr[i] != arr[correct]) {
                swap(arr, i, correct);
            } else {
                i++;
            }
        }

    }

    static void swap(int[] arr, int i, int correct) {
        int temp = arr[i];
        arr[i] = arr[correct];
        arr[correct] = temp;
    }

}

File: python/Cyclic.py
from turtle import circle
from typing import List


class Cycle:
    def sort(arr: List) -> None:
        i = 0
        while (i < len(arr)):
            correct = arr[i]-1
            if (arr[i] == arr[correct]):
                i += 1
            else:
                arr[i], arr[correct] = arr[correct], arr[i]


arr = [3, 1, 4, 2]
Cycle.sort(arr)
print(arr)
